api-compose-object.go

```diff
@@ -458,7 +458,12 @@ func (c Client) ComposeObjectWithProgress(dst DestinationInfo, srcs []SourceInfo
 	for i, src := range srcs {
 		h := src.Headers
 		if src.encryption != nil {
-			encrypt.SSECopy(src.encryption).Marshal(h)
+			if src.encryption.Type() == encrypt.SSEC {
```

Comment:
@poornas This is not necessary AFAIK. See doc and impl of: [`SSECopy`](https://github.com/minio/minio-go/blob/master/pkg/encrypt/server-side.go#L143).

Reply:
Agree the code looks same to me

Reply:
you are correct @aead